menas/src/main/scala/za/co/absa/enceladus/menas/repositories/VersionedMongoRepository.scala

```diff
@@ -41,12 +42,19 @@ abstract class VersionedMongoRepository[C <: VersionedModel](mongoDb: MongoDatab
   private def getParent(oldEntity: C): MenasReference = {
     MenasReference(collection = Some(collectionBaseName), name = oldEntity.name, version = oldEntity.version)
   }
-  
+
+  def distinctCount(): Future[Int] = {
```

Comment:
How about some tests (at least add a couple case to `DatasetRepositoryIntegrationSuite`)?